persistent/Database/Persist/SqlBackend/Internal.hs

```diff
@@ -69,7 +70,7 @@ data SqlBackend = SqlBackend
     -- When left as 'Nothing', we default to using 'defaultPutMany'.
     --
     -- @since 2.8.1
-    , connStmtMap :: IORef (Map Text Statement)
+    , connStmtMap :: StatementCache
```

Comment:
i really like this abstraction, should make it a lot easier to extend and improve things